src/test/java/com/wikia/webdriver/elements/mercury/pages/CategoryPage.java

```diff
@@ -0,0 +1,181 @@
+    member.click();
+
+    new Loading(driver).handleAsyncPageReload();
+
+    String expectedUrl = "/wiki/" + name.replaceAll(" ", "_");
+    String currentUrl = driver.getCurrentUrl();
+    Assertion.assertTrue(
+        currentUrl.contains(expectedUrl),
+        String.format("Expected part \"%s\" was not found in \"%s\".", expectedUrl, currentUrl)
+    );
+    PageObjectLogging.logInfo(String.format("You were redirected to page: \"%s\".", name));
+
+    return this;
+  }
+
+  private CategoryPage articleContainerIsVisible() {
+    wait.forElementVisible(article);
+    PageObjectLogging.logInfo("Article container is visible.");
+
+    return this;
+  }
+
+  private CategoryPage articleContainerIsNotPresent() {
+    wait.forElementNotPresent(article);
+    PageObjectLogging.logInfo("Article container is not present in DOM, which is nice.");
+
+    return this;
+  }
+
+  private CategoryPage categorySectionsContainerIsVisible() {
+    wait.forElementVisible(categorySections);
+    PageObjectLogging.logInfo("Category sections container is visible.");
+
+    return this;
+  }
+
+  private CategoryPage noMembersMessageIsVisible() {
+    wait.forElementVisible(noMembersMessage);
+    PageObjectLogging.logInfo("Info message about no pages in category is visible.");
+
+    return this;
+  }
+
+  private CategoryPage navigateThroughSection(String name, By buttonSelector) {
+    WebElement section = driver.findElement(
+        By.cssSelector(String.format("#%s.category-section", name))
+    );
+    wait.forElementVisible(section);
+    PageObjectLogging.logInfo(String.format("Category section \"%s\" is visible.", name));
+
+    WebElement sectionBatchFirstItem = section.findElement(By.cssSelector("li"));
+    wait.forElementVisible(sectionBatchFirstItem);
+    String firstItemTextInFirstBatch = sectionBatchFirstItem.getText();
+
+    WebElement buttonElement = section.findElement(buttonSelector);
+    wait.forElementClickable(buttonElement);
+    buttonElement.click();
+    waitForBatchToBeLoaded(section);
+
+    sectionBatchFirstItem = section.findElement(By.cssSelector("li"));
+    wait.forElementVisible(sectionBatchFirstItem);
+    String firstItemTextInSecondBatch = sectionBatchFirstItem.getText();
+
+    Assertion.assertNotEquals(
+        firstItemTextInFirstBatch,
+        firstItemTextInSecondBatch,
+        "First element of the newly loaded section is the same as the previous one."
+    );
+    PageObjectLogging.logInfo("New section was loaded correctly.");
+
+    return this;
+  }
+
+  private void waitForBatchToBeLoaded(WebElement section) {
+    int attemptLimit = 5;
+
+    PageObjectLogging.logInfo("Waiting for loading-batch to be present");
+
+    for (int i = 1; i <= attemptLimit; ++i) {
+      if (section.getAttribute("class").contains("loading-batch")) {
+        PageObjectLogging.logInfo("loading-batch class was present in attempt: " + i + " of " + attemptLimit);
+
+        break;
+      }
+
+      try {
+        Thread.sleep(200);
```

Comment:
https://selenium.googlecode.com/git/docs/api/java/org/openqa/selenium/support/ui/FluentWait.html and talk to @ludwikkazmierczak